test/testother.cpp

```diff
@@ -6867,6 +6867,31 @@ class TestOther : public TestFixture {
                       "[test.cpp:4]: (style) The comparison 'E0 > 0' is always false.\n"
                       "[test.cpp:5]: (style) The comparison 'E0 == 0' is always true.\n",
                       errout.str());
+
+        check("struct S {\n" // #12040, #12044
+              "    static const int I = 0;\n"
+              "    enum { E0 };\n"
+              "    enum F { F0 };\n"
+              "    void f() {\n"
+              "        if (0 > I) {}\n"
+              "        if (0 > S::I) {}\n"
+              "        if (0 > E0) {}\n"
+              "        if (0 > S::E0) {}\n"
```

Comment:
Is valueflow updated for these cases?

Reply:
I think so. The fix enables `isSameConstantValue()` to do its thing.
~~~
##Value flow
Line 2
  I always 0
  = always 0
  0 always 0
Line 3
  E0 always 0
Line 4
  F0 always 0
Line 6
  0 always 0
  > always 0
  I always 0
Line 7
  0 always 0
  > always 0
  :: always 0
  I always 0
Line 8
  0 always 0
  > always 0
  E0 always 0
Line 9
  0 always 0
  > always 0
  :: always 0
  E0 always 0
Line 13
  0 always 0
  > always 0
  :: always 0
  I always 0
Line 14
  0 always 0
  > always 0
  :: always 0
  E0 always 0
Line 15
  0 always 0
  > always 0
  :: always 0
  F0 always 0
~~~